### ISSUE TITLE
Deprecations for Guard 3.0

### ISSUE THREAD

**Issue by e2** [guard/guard#605]
TL;DR - these will be changed/removed from Guard 3.0

It's getting too time consuming to refactor, audit, and extend the current version of Guard, so I might have to switch to 3.0 just because a few changes may break things, so ...

I'm creating a list here of things to remove and create Guard 3.0 (just to break backward compatibility, but make it easier to contribute by removing obsolete/unused/deprecated methods).
#### What kind of changes qualify?

Those that make it trivial to migrate from 2.x and don't require a lot for work to implement/remove/change and/or break backward compatibility, but don't really affect users.

Any suggestions (especially for removals) are most welcome.
